src/browser-connection/index.js

```diff
@@ -48,18 +48,19 @@ export default class BrowserConnection extends EventEmitter {
         }, this.HEARTBEAT_TIMEOUT);
     }
 
-    _popNextTestRunUrl () {
-        if (this.jobQueue.length) {
-            var job = this.jobQueue[0];
-            var url = job.popNextTestRunUrl();
+    _hasQueuedJobs () {
+        return !!this.jobQueue.length;
+    }
 
-            if (!job.hasQueuedTestRuns)
-                this.jobQueue.shift();
+    _currentJob () {
```

Comment:
Both `_hasQueuedJobs` and `_currentJob` can be getters like `hasQueuedTestRuns` in `BrowserJob`